actors/builtin/market/market_actor.go

```diff
@@ -467,7 +529,7 @@ func (a Actor) CronTick(rt Runtime, params *adt.EmptyValue) *adt.EmptyValue {
 					// Not yet appeared in proven sector; check for timeout.
 					AssertMsg(rt.CurrEpoch() >= deal.StartEpoch, "if sector start is not set, we must be in a timed out state")
 
-					slashed := st.processDealInitTimedOut(rt, et, lt, dealID, deal, state)
+					slashed := st.processDealInitTimedOut(rt, et, lt, pending, dealID, deal, state)
```

Comment:
In combination with my comment above, we need to delete from pending regardless of whether is has a sector start epoch.